src/transformers/generation/stopping_criteria.py

```diff
@@ -155,3 +246,90 @@ def validate_stopping_criteria(stopping_criteria: StoppingCriteriaList, max_leng
     elif stopping_max_length is None:
         new_stopping_criteria.append(MaxLengthCriteria(max_length=max_length))
     return new_stopping_criteria
+
+
+def _stop_string_get_matching_positions(
```

Comment:
I'd add a comment explaining why we have two external functions, on each function (this one for testing purposes, the other one for caching purposes)